test/servers/binary_framed_test.exs

```diff
@@ -75,11 +75,13 @@ defmodule Servers.BinaryFramedIntegrationTest do
   alias Thrift.TApplicationException, as: TAE
 
   def stop_server(server_pid) do
-    ref = Process.monitor(server_pid)
-    Server.Framed.stop(server_pid)
-    receive do
-      {:DOWN, ^ref, _, _, _} ->
-        :ok
+    if Process.alive?(server_pid) do
+      ref = Process.monitor(server_pid)
+      Server.Framed.stop(server_pid)
```

Comment:
There's a race condition. The process could be alive when we call `Process.alive?` but dead when we get to the `Server.Framed.stop`. You could catch the noproc though.
```elixir
ref = Process.monitor(server_pid)
try do
  Server.Framed.stop(server_pid)
catch :exit, :noproc -> :ok
end
assert_receive {:DOWN, ^ref, _, _, _}, 5000
```